ops/instrument_httpclient_test.go

```diff
@@ -0,0 +1,80 @@
+package ops
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/valyala/fasthttp"
+
+	"github.com/valkyrie-fnd/valkyrie/rest"
+)
+
+type mockPipelineContext[T any] struct {
+	ctx     context.Context
+	payload T
+}
+
+func (m *mockPipelineContext[T]) Next() error {
+	return nil
+}
+
+func (m *mockPipelineContext[T]) Context() context.Context {
+	return m.ctx
+}
+
+func (m *mockPipelineContext[T]) SetContext(ctx context.Context) {
+	m.ctx = ctx
+}
+
+func (m *mockPipelineContext[T]) Payload() T {
+	return m.payload
+}
+
+func Test_httpTracingHandler(t *testing.T) {
+	handler := httpTracingHandler()
```

Comment:
It turned out very cumbersome to assert that traces were actually produced, so for now these tests are just rudimentary checks that no errors happen